### PR TITLE
Add agency settings toggle

### PR DESCRIPTION
**Pull request by lsl**
Fixes https://github.com/Automattic/dotcom-forge/issues/9610

## Proposed Changes

- Adds the agency toggle
- Had to diverge from mock up to re-use settings form
- Most sites won't show this option, to test either use an a4a site or swap the toggles around in code to display.

Screenshot 2024-10-31 at 17-20-17 Sites ‹ Site Title — WordPress com

### CODE REVIEW

**Comment by lsl** [comment]
e.g. testing patch:

```diff
diff --git a/client/my-sites/site-settings/a4a-fully-managed-site-setting.tsx b/client/my-sites/site-settings/a4a-fully-managed-site-setting.tsx
index 5187c3ac3d0..930c8849095 100644
--- a/client/my-sites/site-settings/a4a-fully-managed-site-setting.tsx
+++ b/client/my-sites/site-settings/a4a-fully-managed-site-setting.tsx
@@ -32,7 +32,7 @@ export function A4AFullyManagedSiteSetting( {
 
        const shouldShowToggle = agencySite && isAtomicSite;
 
-       if ( ! shouldShowToggle ) {
+       if ( shouldShowToggle ) {
                return null;
        }
 
diff --git a/client/sites/settings/agency/index.jsx b/client/sites/settings/agency/index.jsx
index 1d5ae4c1a85..92706b38e42 100644
--- a/client/sites/settings/agency/index.jsx
+++ b/client/sites/settings/agency/index.jsx
@@ -23,7 +23,7 @@ const AgencySettings = ( {
        return (
                <div className="agency-settings">
                        <NavigationHeader title={ __( 'Agency' ) } />
-                       { shouldShowToggle && (
+                       { ! shouldShowToggle && (
                                <A4AFullyManagedSiteSetting
                                        title={ __( 'Client Access' ) }
                                        site={ site }
@@ -34,7 +34,7 @@ const AgencySettings = ( {
                                        disabled={ isRequestingSettings || isSavingSettings }
                                />
                        ) }
-                       { ! shouldShowToggle && (
+                       { shouldShowToggle && (
                                <div>
                                        <p>{ __( 'Agency settings are not available for this site.' ) }</p>
                                </div>
diff --git a/client/sites/settings/controller.tsx b/client/sites/settings/controller.tsx
index bd94e9a2b39..a4391673f61 100644
--- a/client/sites/settings/controller.tsx
+++ b/client/sites/settings/controller.tsx
@@ -31,7 +31,7 @@ export function SettingsSidebar() {
                                        { __( 'Administration' ) }
                                </SidebarItem>
                        ) }
-                       { shouldShowAgency && (
+                       { ! shouldShowAgency && (
                                <SidebarItem href={ `/sites/settings/agency/${ slug }` }>{ __( 'Agency' ) }</SidebarItem>
                        ) }
                        <SidebarItem href={ `/sites/settings/caches/${ slug }` }>{ __( 'Caches' ) }</SidebarItem>
```

File: client/sites/settings/agency/index.jsx [hunk 1]
```diff
@@ -1,7 +1,58 @@
...
+	handleToggle,
+	isRequestingSettings,
+	isSavingSettings,
+} ) => {
+	const site = useSelector( getSelectedSite );
+	const isAtomicSite = site.is_wpcom_atomic;
+
+	const { data: agencySite } = useFetchAgencyFromBlog( site?.ID, { enabled: !! site?.ID } );
+
+	const shouldShowToggle = agencySite && isAtomicSite;
```
Comment (fushar) [inline client/sites/settings/agency/index.jsx]: Hmm... I think we need to start thinking about this redundant guard. It makes the code a bit convoluted just to accommodate a very edge case. The page will be broken anyway due to the lack of active sidebar item.

I think we can do something like 

```jsx
<SidebarItem enabled={shouldShowAgency} ...>
```

and in `SidebarItem`, we just throw 404 / redirect to site selection. What do you think?

Comment (fushar) [reply client/sites/settings/agency/index.jsx]: Proposing: https://github.com/Automattic/wp-calypso/pull/95937

Comment (lsl) [reply client/sites/settings/agency/index.jsx]: Left some comments, agree there's an issue here not quite sure what the best approach is yet.

**Review by fushar — APPROVED** [review]
👍 Let's iterate later.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16957222

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @lsl for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16957222).

**Pull request merged**